### ISSUE TITLE
teleport sessions lock up  when running many tsh session in parallel

### ISSUE THREAD

**Issue by slavitch** [gravitational/teleport#13088]
Expected behavior:

Ability to run dozens of simultaneous tsh sessions in parallel in the background
Current behavior:

Things grind to a halt if you run more than 2 or 3 at the same time.

Bug details:
- Teleport version 9.3.2
- Recreation steps  run a bot service that creates multiple parallel tsh sessions

**Comment by slavitch** [comment]
DEBU [CLIENT]    Updated window size from (211, 55) to (211, 57) due to SIGWINCH. client/session.go:427

**Comment by zmb3** [comment]
This only happens when using tbot? Or do you see the same behavior if you run several concurrent tsh sessions as a regular user?

**Comment by zmb3** [comment]
Log items included in another ticket:


Looks like some network timeouts:

```
Jun  1 23:34:28 ip-172-31-20-73 teleport[38599]: User Message: readfrom tcp 172.31.20.73:60498-&gt;172.31.20.73:3025: read tcp 172.31.20.73:443-&gt;24.212.231.149:44332: read: connection reset by peer] alpnproxy/proxy.go:322
Jun  1 23:34:33 ip-172-31-20-73 teleport[38599]: 2022-06-01T23:34:33Z WARN [MX:PROXY:] "\nERROR REPORT:\nOriginal Error: *net.OpError read tcp 172.31.20.73:443-&gt;192.241.222.79:46322: i/o timeout\nStack Trace:\n\t/go/src/github.com/gravitational/teleport/lib/multiplexer/multiplexer.go:261 github.com/gravitational/teleport/lib/multiplexer.detect\n\t/go/src/github.com/gravitational/teleport/lib/multiplexer/multiplexer.go:221 github.com/gravitational/teleport/lib/multiplexer.(*Mux).detectAndForward\n\t/opt/go/src/runtime/asm_amd64.s:1581 runtime.goexit\nUser Message: failed to peek connection\n\tread tcp 172.31.20.73:443-&gt;192.241.222.79:46322: i/o timeout" multiplexer/multiplexer.go:224
Jun  1 23:34:36 ip-172-31-20-73 teleport[38599]: 2022-06-01T23:34:36Z WARN [MX:PROXY:] "\nERROR REPORT:\nOriginal Error: *net.OpError read tcp 172.31.20.73:443-&gt;38.81.198.66:40884: i/o timeout\nStack Trace:\n\t/go/src/github.com/gravitational/teleport/lib/multiplexer/multiplexer.go:261 github.com/gravitational/teleport/lib/multiplexer.detect\n\t/go/src/github.com/gravitational/teleport/lib/multiplexer/multiplexer.go:221 github.com/gravitational/teleport/lib/multiplexer.(*Mux).detectAndForward\n\t/opt/go/src/runtime/asm_amd64.s:1581 runtime.goexit\nUser Message: failed to peek connection\n\tread tcp 172.31.20.73:443-&gt;38.81.198.66:40884: i/o timeout" multiplexer/multiplexer.go:224
Jun  1 23:34:36 ip-172-31-20-73 teleport[38599]: 2022-06-01T23:34:36Z WARN [MX:PROXY:] "\nERROR REPORT:\nOriginal Error: *net.OpError read tcp 172.31.20.73:443-&gt;38.81.198.36:5253: i/o timeout\nStack Trace:\n\t/go/src/github.com/gravitational/teleport/lib/multiplexer/multiplexer.go:261 github.com/gravitational/teleport/lib/multiplexer.detect\n\t/go/src/github.com/gravitational/teleport/lib/multiplexer/multiplexer.go:221 github.com/gravitational/teleport/lib/multiplexer.(*Mux).detectAndForward\n\t/opt/go/src/runtime/asm_amd64.s:1581 runtime.goexit\nUser Message: failed to peek connection\n\tread tcp 172.31.20.73:443-&gt;38.81.198.36:5253: i/o timeout" multiplexer/multiplexer.go:224
```

As well as some warnings from the ALPN proxy:

```
Jun  1 21:46:52 ip-172-31-20-73 teleport[38599]: User Message: readfrom tcp 172.31.20.73:52788-&gt;172.31.20.73:3025: read tcp 172.31.20.73:443-&gt;69.70.215.194:41372: read: connection reset by peer] alpnproxy/proxy.go:322
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: 2022-06-01T21:47:11Z WARN [ALPN:PROX] Failed to handle client connection. error:[
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: ERROR REPORT:
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: Original Error: trace.aggregate readfrom tcp 172.31.20.73:52814-&gt;172.31.20.73:3025: read tcp 172.31.20.73:443-&gt;24.212.231.149:44216: read: connection reset by peer
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: Stack Trace:
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: #011/go/src/github.com/gravitational/teleport/lib/srv/alpnproxy/auth/auth_proxy.go:180 github.com/gravitational/teleport/lib/srv/alpnproxy/auth.(*AuthProxyDialerService).proxyConn
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: #011/go/src/github.com/gravitational/teleport/lib/srv/alpnproxy/auth/auth_proxy.go:73 github.com/gravitational/teleport/lib/srv/alpnproxy/auth.(*AuthProxyDialerService).HandleConnection
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: #011/go/src/github.com/gravitational/teleport/lib/srv/alpnproxy/proxy.go:209 github.com/gravitational/teleport/lib/srv/alpnproxy.(*HandlerDecs).handle
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: #011/go/src/github.com/gravitational/teleport/lib/srv/alpnproxy/proxy.go:368 github.com/gravitational/teleport/lib/srv/alpnproxy.(*Proxy).handleConn
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: #011/go/src/github.com/gravitational/teleport/lib/srv/alpnproxy/proxy.go:314 github.com/gravitational/teleport/lib/srv/alpnproxy.(*Proxy).Serve.func1
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: #011/opt/go/src/runtime/asm_amd64.s:1581 runtime.goexit
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: User Message: readfrom tcp 172.31.20.73:52814-&gt;172.31.20.73:3025: read tcp 172.31.20.73:443-&gt;24.212.231.149:44216: read: connection reset by peer] alpnproxy/proxy.go:322
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: 2022-06-01T21:47:11Z WARN [ALPN:PROX] Failed to handle client connection. error:[
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: ERROR REPORT:
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: Original Error: trace.aggregate readfrom tcp 172.31.20.73:52816-&gt;172.31.20.73:3025: read tcp 172.31.20.73:443-&gt;24.212.231.149:44220: read: connection reset by peer
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: Stack Trace:
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: #011/go/src/github.com/gravitational/teleport/lib/srv/alpnproxy/auth/auth_proxy.go:180 github.com/gravitational/teleport/lib/srv/alpnproxy/auth.(*AuthProxyDialerService).proxyConn
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: #011/go/src/github.com/gravitational/teleport/lib/srv/alpnproxy/auth/auth_proxy.go:73 github.com/gravitational/teleport/lib/srv/alpnproxy/auth.(*AuthProxyDialerService).HandleConnection
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: #011/go/src/github.com/gravitational/teleport/lib/srv/alpnproxy/proxy.go:209 github.com/gravitational/teleport/lib/srv/alpnproxy.(*HandlerDecs).handle
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: #011/go/src/github.com/gravitational/teleport/lib/srv/alpnproxy/proxy.go:368 github.com/gravitational/teleport/lib/srv/alpnproxy.(*Proxy).handleConn
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: #011/go/src/github.com/gravitational/teleport/lib/srv/alpnproxy/proxy.go:314 github.com/gravitational/teleport/lib/srv/alpnproxy.(*Proxy).Serve.func1
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: #011/opt/go/src/runtime/asm_amd64.s:1581 runtime.goexit
Jun  1 21:47:11 ip-172-31-20-73 teleport[38599]: User Message: readfrom tcp 172.31.20.73:52816-&gt;172.31.20.73:3025: read tcp 172.31.20.73:443-&gt;24.212.231.149:44220: read: connection reset by peer] alpnproxy/proxy.go:322
```

**Comment by eloekset** [comment]
I get the same error when enrolling a new Debian PC:
![image](https://user-images.githubusercontent.com/791721/235322762-dd032353-9569-4600-9a91-405b4457abbc.png)
![image](https://user-images.githubusercontent.com/791721/235322767-855f1531-5a8e-44cc-b97e-8e7239ecab00.png)

The PC seems to think everything is good:
![image](https://user-images.githubusercontent.com/791721/235322812-6627ee39-4bd7-41d9-9137-1204b5121554.png)

I'm running Teleport on Proxmox behind a Traefik controller. I had issues with the Teleport setup when acme should receive a certificate, so I disabled acme in the teleport.yaml file and started using teleport with the certificate that Traefik had already setup.
![image](https://user-images.githubusercontent.com/791721/235322929-f948f3e5-8191-4686-98c6-79756d14c96c.png)

I'm not able to log in to the demo application that the installation guide tells us to setup:
![image](https://user-images.githubusercontent.com/791721/235322996-9ad03ff8-7f90-47bf-bc54-aae1de7f376e.png)
